### ISSUE TITLE
Using a simple import

### ISSUE THREAD

**Issue by donno2048** [donno2048/restricted-functions#6]
Okay, I kinda figured it out, the problem is that the `__builtins__` variable points to the `builtins` module in the main module, but to the private `__builtins__.__dict__` variable in the imported module, I need to find a way around this and then it's supposed to work,

BTW, the simple

```py
if __name__ == '__main__': __builtins__.__dict__['__import__'] = importer
else: builtins['__import__'] = importer
```

won't work either, it will just modify the local builtins

**Comment by iam-py-test** [comment]
I will try to get some work on this, but I am confused. 
How did you even discover how to get into Pythons workings?

**Comment by donno2048** [comment]
> How did you even discover how to get into Pythons workings?

I honestly don't understand what that supposed to mean, I guess you're asking how I know how does the `__builtins__` works?

**Comment by iam-py-test** [comment]
> > How did you even discover how to get into Pythons workings?
> 
> I honestly don't understand what that supposed to mean, I guess you're asking how I know how does the `__builtins__` works?

Yes. Thanks!

**Comment by donno2048** [comment]
I read the Python source code and am a python contributor

**Comment by iam-py-test** [comment]
> I read the Python source code and am a python contributor

Cool! Would you mind posting a link?

**Comment by donno2048** [comment]
To the source code of python?
https://github.com/python/cpython

**Comment by donno2048** [comment]
I will add a branch to make this a package but won't marge it until there is a solution to this problem, do you want to pick a package name and an import name?

**Comment by donno2048** [comment]
I'll make the package name `restricted-functions` and the import name `ref` for now

**Comment by donno2048** [comment]
There is a simple solution, we can just pass the `__builtins__` object **with the dict classifier** in the main module, but it's kinda ugly, what do you think?

**Comment by iam-py-test** [comment]
Sorry I did not respond; I had walked away

**Comment by donno2048** [comment]
> Sorry I did not respond; I had walked away

It's fine, I had projects where I got no response for a week or so...

You don't need to be available all the time

**Comment by iam-py-test** [comment]
The package name sounds good. I am not sure what should be done with the `__builtins__`, but as the simple solution works, I think we should use it for now.

> It's fine, I had projects where I got no response for a week or so...

If I am working on something, I try to check for responses fairly often. 

Has os.popen already been added? If not, I think it should, even thought it is decrepitated.

**Comment by donno2048** [comment]
> The package name sounds good. I am not sure what should be done with the `__builtins__`, but as the simple solution works, I think we should use it for now.

It works now because it's not a module, as a module you cannot just read the files,

We could make it such that you need to initialize it like that:

```py
import importer
importer.main(__builtins__)
```

**Comment by donno2048** [comment]
> Is os.popen already been added? If not, I think it should, even thought it is decrepitated.

You're right, it is even "stronger" than `os.system` so...

**Comment by iam-py-test** [comment]
> > The package name sounds good. I am not sure what should be done with the `__builtins__`, but as the simple solution works, I think we should use it for now.
> 
> It works now because it's not a module, as a module you cannot just read the files,
> 
> We could make it such that you need to initialize it like that:
> 
> ```python
> import importer
> importer.main(__builtins__)
> ```

That might work, as it would also let the user chose when they blocked the imports and would allow for security levels to be passed in as args.

**Comment by donno2048** [comment]
> > > The package name sounds good. I am not sure what should be done with the `__builtins__`, but as the simple solution works, I think we should use it for now.
> > 
> > 
> > It works now because it's not a module, as a module you cannot just read the files,
> > We could make it such that you need to initialize it like that:
> > ```python
> > import importer
> > importer.main(__builtins__)
> > ```
> 
> That might work, as it would also let the user chose when they blocked the imports

I'll do it like that then

(It's not a bug it's a feature)

**Comment by donno2048** [comment]
Fixed in #8
